pkg/controller/atlasproject/backupcompliancepolicy.go

```diff
@@ -0,0 +1,151 @@
+		compliancePolicy.Annotations = map[string]string{}
+	}
+	if projectIds, ok := compliancePolicy.Annotations[ProjectAnnotation]; !ok {
+		compliancePolicy.Annotations[ProjectAnnotation] = project.ID()
+	} else {
+		if !slices.Contains(strings.Split(projectIds, ","), project.ID()) {
+			compliancePolicy.Annotations[ProjectAnnotation] = projectIds + "," + project.ID()
+		}
+	}
+
+	err = r.Client.Update(ctx.Context, compliancePolicy)
+	if err != nil {
+		ctx.Log.Errorf("failed to update backup compliance policy: %v", err)
+		return workflow.Terminate(workflow.ProjectBackupCompliancePolicyUnavailable, err.Error())
+	}
+
+	// check if compliance policy exists in atlas (and matches)
+	// if match, return workflow.OK()
+	atlasCompliancePolicy, _, err := ctx.SdkClient.CloudBackupsApi.GetDataProtectionSettings(ctx.Context, project.ID()).Execute()
+
+	if err != nil {
+		ctx.Log.Errorf("failed to get backup compliance policy from atlas: %v", err)
+		return workflow.Terminate(workflow.ProjectBackupCompliancePolicyUnavailable, err.Error())
+	}
+
+	// otherwise, create/update compliance policy...
+	// create compliance policy in atlas
+	result := syncBackupCompliancePolicy(ctx, project.ID(), *compliancePolicy, *atlasCompliancePolicy)
+	ctx.SetConditionFromResult(status.BackupComplianceReadyType, result)
+	return result
+}
+
+func IsBackupComplianceEmpty(backupCompliancePolicyRef *common.ResourceRefNamespaced) bool {
+	return (backupCompliancePolicyRef == nil) || (backupCompliancePolicyRef.Name == "")
+}
+
+// syncBackupCompliancePolicy compares the compliance policy specified in Kubernetes to the one currently present in Atlas, updating Atlas should they differ.
+func syncBackupCompliancePolicy(ctx *workflow.Context, groupID string, kubeCompliancePolicy mdbv1.AtlasBackupCompliancePolicy, atlasCompliancePolicy admin.DataProtectionSettings20231001) workflow.Result {
+	// convert the CR type to atlas type, so we can compare
+	localCompliancePolicy := kubeCompliancePolicy.ToAtlas(groupID)
+	// sort the slices, so we can compare
+	if localCompliancePolicy.ScheduledPolicyItems != nil {
+		slices.SortFunc(*localCompliancePolicy.ScheduledPolicyItems, compareSPI)
+	}
+	if atlasCompliancePolicy.ScheduledPolicyItems != nil {
+		slices.SortFunc(*atlasCompliancePolicy.ScheduledPolicyItems, compareSPI)
+	}
+	// deep equal, now that the slices are sorted
+	if !equality.Semantic.DeepEqual(localCompliancePolicy, atlasCompliancePolicy) {
+		_, _, err := ctx.SdkClient.CloudBackupsApi.UpdateDataProtectionSettings(ctx.Context, groupID, localCompliancePolicy).Execute()
+		if err != nil {
+			ctx.Log.Errorf("failed to update backup compliance policy in atlas: %v", err)
+			return workflow.Terminate(workflow.ProjectBackupCompliancePolicyNotCreatedInAtlas, err.Error())
+		}
+	}
+	return workflow.OK()
+}
+
+// compareSPI is a function for deciding if one instance is less than another, for use when sorting
+func compareSPI(a, b admin.BackupComplianceScheduledPolicyItem) int {
```

Comment:
rather than doing this here on the Atlas SDK type we should do this on our CRD types, as it is a semantically important functionality:

```
package v1

type AtlasBackupPolicyItems []AtlasBackupPolicyItem

func (a AtlasBackupPolicyItems) Len() int      { return len(a) }
func (a AtlasBackupPolicyItems) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
func (a AtlasBackupPolicyItems) Less(i, j int) bool {
	ai, aj := a[i], a[j]
	switch {
	case ai.FrequencyType != aj.FrequencyType:
		return ai.FrequencyType < aj.FrequencyType
	case ai. FrequencyInterval != aj.FrequencyInterval:
		return ai.FrequencyInterval < aj.FrequencyInterval
	case ai.RetentionUnit != aj.RetentionUnit:
		return ai.RetentionUnit < aj.RetentionUnit
	default:
		return a.RetentionValue < b.RetentionValue
	}
}

func (s *AtlasBackupCompliancePolicySpec) Normalize() {
	sort.Sort(AtlasBackupPolicyItems(s.ScheduledPolicyItems))
}
```